### DISCUSSION TITLE
mHM Optimizers

### DISCUSSION THREAD

**Discussion by StephanThober** [mhm-ufz/mHM#3] [Q&A] [answered]
Hi all,

which optimization algorithms are contained in the mHM model?

Thanks,
Stephan

**Comment by MuellerSeb** [comment, marked as answer]
Hi there,

thanks for using mHM. The provided optimization methods are:

- `0` Markov chain Monte Carlo (MCMC)
- `1` Dynamically Dimensioned Search (DDS) 
- `2` Simulated Annealing (SA)
- `3` Shuffled Complex Evolution algorith (SCE)

they can be selected by the `opti_method` variable in `mhm.nml`.

The target function for optimization can be selected by `opti_function`. See the documentation from `mhm.nml`:

- `1`  SO: Q:   1.0 - NSE
- `2`  SO: Q:   1.0 - lnNSE
- `3`  SO: Q:   1.0 - 0.5*(NSE+lnNSE)
- `4`  SO: Q:  -1.0 * loglikelihood with trend removed from absolute errors and then lag(1)-autocorrelation removed
- `5`  SO: Q:   ((1-NSE)**6+(1-lnNSE)**6)**(1/6)
- `6`  SO: Q:   SSE
- `7`  SO: Q:  -1.0 * loglikelihood with trend removed from absolute errors
- `8`  SO: Q:  -1.0 * loglikelihood with trend removed from the relative errors and then lag(1)-autocorrelation removed
- `9`  SO: Q:  1.0 - KGE (Kling-Gupta efficiency measure)
- `10` SO: SM: 1.0 - KGE of catchment average soilmoisture
- `11` SO: SM: 1.0 - Pattern dissimilarity (PD) of spatially distributed soil moisture
- `12` SO: SM: Sum of squared errors (SSE) of spatially distributed standard score (normalization) of soil moisture
- `13` SO: SM: 1.0 - average temporal correlation of spatially distributed soil moisture
- `14` SO: Q:  sum[((1.0-KGE_i)/ nGauges)**6]**(1/6) > combination of KGE of every gauging station based on a power-6 norm
- `15` SO: Q + domain_avg_TWS: [1.0-KGE(Q)]*RMSE(domain_avg_TWS) - objective function using Q and domain average (standard score) TWS
- `16) (reserved) please use the next number when implementing a new one
      MO: Q:  1st objective: (1) = 1.0 - NSE
          Q:  2nd objective: (2) = 1.0 - lnNSE
- `17` SO: N:  1.0 - KGE of spatio-temporal neutron data, catchment-average
- `18` (reserved) please use the next number when implementing a new one
      MO: Q:  1st objective: 1.0 - lnNSE(Q_highflow)  (95% percentile)
          Q:  2nd objective: 1.0 - lnNSE(Q_lowflow)   (5% of data range)
- `19` (reserved) please use the next number when implementing a new one
      MO: Q:  1st objective: 1.0 - lnNSE(Q_highflow)  (non-low flow)
          Q:  2nd objective: 1.0 - lnNSE(Q_lowflow)   (5% of data range)
- `20` (reserved) please use the next number when implementing a new one
      MO: Q:  1st objective: absolute difference in FDC's low-segment volume
          Q:  2nd objective: 1.0 - NSE of discharge of months DJF
- `21` (reserved) please use the next number when implementing a new one
      SO: Q:  ( (1.0-lnNSE(Q_highflow))**6 + (1.0-lnNSE(Q_lowflow))**6 )**(1/6)
              where Q_highflow and Q_lowflow are calculated like in objective (19)
- `22-26` (reserved MC/JM/ST) please use the next number when implementing a new one
- `27` SO: ET: 1.0 - KGE of catchment average evapotranspiration
- `28` SO: Q + SM: weighted OF using SM (OF12) and Q (OF14) equally weighted
- further functions can be implemented in mo_objective_function and mo_mrm_objective_function
- `29` SO: Q + ET: weighted OF using ET (OF27) and Q (OF14) equally weighted
- `30` SO: Q + domain_avg_ET: [1.0-KGE(Q)]*RMSE(domain_avg_ET) - objective function using Q and domain average ET (standard score)$
- `31` SO: Q: 1  - weighted NSE (NSE is weighted with observed discharge)

further functions can be implemented in mo_objective_function and mo_mrm_objective_function

You can read more about that in the documentation: https://mhm.pages.ufz.de/mhm/stable/calopt.html

Hope that helps!

Cheers, Sebastian
